seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/JobExecutionIT.java

```diff
@@ -129,6 +130,32 @@ public void cancelJobTest() throws Exception {
                                                         objectCompletableFuture.get())));
     }
 
+    @Test
+    public void testExpiredJobWasDeleted() throws Exception {
+        Common.setDeployMode(DeployMode.CLIENT);
+        String filePath = TestUtils.getResource("batch_fakesource_to_file.conf");
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("job_expire");
+
+        ClientConfig clientConfig = ConfigProvider.locateAndGetClientConfig();
+        clientConfig.setClusterName(TestUtils.getClusterName("JobExecutionIT"));
+        SeaTunnelClient engineClient = new SeaTunnelClient(clientConfig);
+        JobExecutionEnvironment jobExecutionEnv =
+                engineClient.createExecutionContext(filePath, jobConfig);
+
+        final ClientJobProxy clientJobProxy = jobExecutionEnv.execute();
+
+        JobResult result = clientJobProxy.doWaitForJobComplete().get();
+        Assertions.assertEquals(result.getStatus(), JobStatus.FINISHED);
+        Awaitility.await()
+                .atMost(65, TimeUnit.SECONDS)
+                .untilAsserted(
+                        () ->
+                                Assertions.assertThrowsExactly(
+                                        NullPointerException.class,
```

Comment:
This check is related to this PR https://github.com/apache/seatunnel/pull/5007

Reply:
My test case here is different from #5007, I called it first ClientJobProxy. doWaitForJobComplete () method ensures that the task must be correct and completely without throwing an exception, and then call JobHistoryService obtaining state method, the service side to there is no Job all returns null Pointers, so empty under direct judgment The pointer will confirm that the history Job has been deleted. Personally, I think another state should be added to express that Job does not exist, but I did not add this state here because the influence of the increase or decrease of the state is too wide